exist-core/src/main/java/org/exist/xquery/XQueryContext.java

```diff
@@ -3058,34 +2946,26 @@ public Pragma getPragma(final String name, String contents) throws XPathExceptio
         } catch (final QName.IllegalQNameException e) {
             throw new XPathException(rootExpression, ErrorCodes.XPST0081, "No namespace defined for prefix " + name);
         }
+        final String ns = qname.getNamespaceURI();
+        if (ns.isEmpty()) {
+            throw new XPathException(rootExpression, ErrorCodes.XPST0081, "XPST0081: pragma's ('" + name + "') namespace URI is empty");
+        }
 
-        if (qname.getNamespaceURI().isEmpty()) {
-            throw new XPathException(rootExpression, "XPST0081: pragma's ('" + name + "') namespace URI is empty");
-        } else if (Namespaces.EXIST_NS.equals(qname.getNamespaceURI())) {
-            contents = StringValue.trimWhitespace(contents);
-
-            if (TimerPragma.TIMER_PRAGMA.equals(qname)) {
-                return new TimerPragma(rootExpression, qname, contents);
-            }
-
-            if (Optimize.OPTIMIZE_PRAGMA.equals(qname)) {
-                return new Optimize(rootExpression, this, qname, contents, true);
-            }
-
-            if (ForceIndexUse.EXCEPTION_IF_INDEX_NOT_USED_PRAGMA.equals(qname)) {
-                return new ForceIndexUse(rootExpression, qname, contents);
-            }
-
-            if (ProfilePragma.PROFILING_PRAGMA.equals(qname)) {
-                return new ProfilePragma(rootExpression, qname, contents);
-            }
-
-            if (NoIndexPragma.NO_INDEX_PRAGMA.equals(qname)) {
-                return new NoIndexPragma(rootExpression, qname, contents);
-            }
+        if (!Namespaces.EXIST_NS.equals(ns)) {
+            // TODO(JL): should we throw or a least warn here?
+            return null;
         }
 
-        return null;
+        final String sanitizedContents = StringValue.trimWhitespace(contents);
+
+        return switch(qname.getLocalPart()) {
+            case Optimize.LOCAL_NAME -> new Optimize(rootExpression, this, qname, sanitizedContents, true);
```

Comment:
These variables should not all be renamed to `LOCAL_NAME` otherwise they cannot be statically imported into the same class for use as they will conflict. Please rename them back.

Reply:
I did not rename any variable. They are all added.